### PR TITLE
Add support for deploying Spinnaker in Shared VPC service project on …

### PR DESCRIPTION
**Pull request by duftler**
…shared network/subnet.

### CODE REVIEW

File: scripts/manage/add_missing_properties.sh [hunk 1]
```diff
@@ -22,3 +22,7 @@ export CONFIG_CSR_REPO=\$DEPLOYMENT_NAME-config
 EOL
 
 add_property_if_missing CONFIG_CSR_REPO "$CSR_PROPERTY_DECLARATION"
+
+add_property_if_missing NETWORK_PROJECT "export NETWORK_PROJECT=\$PROJECT_ID"
+add_property_if_missing NETWORK_REFERENCE "export NETWORK_REFERENCE=projects/\$NETWORK_PROJECT/global/networks/\$NETWORK"
+add_property_if_missing SUBNET_REFERENCE "export SUBNET_REFERENCE=projects/\$NETWORK_PROJECT/regions/us-west1/subnetworks/\$SUBNET"
```
[the remark this answers is not in the document]
Comment (zee-ahmed) [reply scripts/manage/add_missing_properties.sh]: I've just tested this myself with a shared VPC and i needed to specify `$REGION`. Without it, the redis instance wouldn't deploy via Cloud Memorystore

File: scripts/install/setup_properties.sh [hunk 2]
```diff
@@ -98,6 +114,37 @@ export HALYARD_VERSION=1.22.1
...
+if [ "$SHARED_VPC_HOST_PROJECT" ]; then
+  cat >> ~/spinnaker-for-gcp/scripts/install/properties <<EOL
+# If you want to use a shared network/subnet from the Shared VPC host project, you'll need to perform
+# these steps prior to running the setup.sh script:
+#   1) Specify the name of the shared network in \$NETWORK up above.
+#   2) Specify the name of the shared subnet in \$SUBNET up above.
+#   3) Specify the Shared VPC host project id ($SHARED_VPC_HOST_PROJECT) in \$NETWORK_PROJECT below.
+#   4) Ensure the subnet referenced by \$SUBNET defines 2 named secondary ranges (one for
+#      pods, and one for clusters).
+#   5) Specify the names of the 2 secondary ranges in \$CLUSTER_SECONDARY_RANGE_NAME and
```
[the remark this answers is not in the document]
Comment (louisjimenez) [reply scripts/install/setup_properties.sh]: On second thought, it is still useful to have this information for users with an existing cluster and subnet.

File: scripts/install/setup_properties.sh [hunk 3]
```diff
@@ -98,6 +114,37 @@ export HALYARD_VERSION=1.22.1
...
+
+if [ "$SHARED_VPC_HOST_PROJECT" ]; then
+  cat >> ~/spinnaker-for-gcp/scripts/install/properties <<EOL
+# If you want to use a shared network/subnet from the Shared VPC host project, you'll need to perform
+# these steps prior to running the setup.sh script:
+#   1) Specify the name of the shared network in \$NETWORK up above.
+#   2) Specify the name of the shared subnet in \$SUBNET up above.
+#   3) Specify the Shared VPC host project id ($SHARED_VPC_HOST_PROJECT) in \$NETWORK_PROJECT below.
+#   4) Ensure the subnet referenced by \$SUBNET defines 2 named secondary ranges (one for
+#      pods, and one for clusters).
```
[the remark this answers is not in the document]
Comment (duftler) [reply scripts/install/setup_properties.sh]: Fixed.

File: scripts/manage/add_missing_properties.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (duftler) [reply scripts/manage/add_missing_properties.sh]: Good catch. Fixed.

**Comment by duftler** [comment]
@louisjimenez PTAL

**Pull request merged**